scripts/generate-functionmap.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+if (PHP_VERSION_ID < 80000) {
+    echo 'This script requires PHP 8.0 or higher';
+    exit(1);
+}
+
+$filename = __DIR__ . '/functionmap.php';
+(new FunctionMapGenerator)->createFunctionMap($filename);
+printf("Created call map in %s\n", $filename);
+
+class FunctionMapGenerator
```

Comment:
I didn't realize PHP allowed a class to be instantiated _before_ it was declared. Was that always the case?

Reply:
I'm not sure when that was added - that was the reason I initially had everything differently. Once I realised this works I decided to put all the logic in a class.

Reply:
It seems that it works since PHP 4, except if the class declaration is in a condition.
https://3v4l.org/NW6YX